// evio-library-language/dist/src/classes/redis-strategy.class.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const evio_library_commons_1 = require("evio-library-commons");
const evio_redis_connection_1 = __importDefault(require("evio-redis-connection"));
class ConcreteStrategyRedis {
    async execute(redisKey) {
        const context = 'ConcreteStrategyRedis';
        const data = await evio_redis_connection_1.default.get(redisKey);
        console.log(`[${context}] - Logs`, {
            message: `Fetching translation from Redis with key ${redisKey}`,
            timestamp: new Date().toISOString(),
        });
        if (!data) {
            throw (0, evio_library_commons_1.NotFound)({
                auth: false,
                code: 'server_translation_not_found',
                message: `[${context}] - Translation not found in Redis for key ${redisKey}`,
            });
        }
        return JSON.parse(data);
    }
}
exports.default = ConcreteStrategyRedis;
