// evio-library-language/dist/src/classes/weblate-strategy.class.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const evio_redis_connection_1 = __importDefault(require("evio-redis-connection"));
const axios_1 = __importDefault(require("axios"));
class ConcreteStrategyWeblate {
    async execute(redisKey, weblateRequest) {
        const context = 'ConcreteStrategyWeblate';
        console.info(`[${context}] - Logs`, {
            host: weblateRequest.host,
            message: `Fetching translation from Weblate`,
            timestamp: new Date().toISOString(),
        });
        const { data } = await axios_1.default.get(weblateRequest.host, {
            headers: weblateRequest.headers,
        });
        if (data) {
            await evio_redis_connection_1.default.set(redisKey, JSON.stringify(data));
            console.info(`[${context}] - Logs`, {
                message: `Translation saved to Redis with key ${redisKey}`,
                timestamp: new Date().toISOString(),
            });
        }
        return data;
    }
}
exports.default = ConcreteStrategyWeblate;
