// evio-library-commons/dist/src/error/error.handler.common.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.newError = exports.NotFound = exports.ServerError = exports.Forbidden = exports.Unauthorized = exports.BadRequest = void 0;
const StatusCodeHttp_1 = require("../enums/StatusCodeHttp");
const newError = (statusCode, error, context) => ({ statusCode, error, context });
exports.newError = newError;
const BadRequest = (error, context) => newError(StatusCodeHttp_1.StatusCodeHttp.BAD_REQUEST, error, context);
exports.BadRequest = BadRequest;
const Unauthorized = (error, context) => newError(StatusCodeHttp_1.StatusCodeHttp.UNAUTHORIZED, error, context);
exports.Unauthorized = Unauthorized;
const NotFound = (error, context) => newError(StatusCodeHttp_1.StatusCodeHttp.NOT_FOUND, error, context);
exports.NotFound = NotFound;
const Forbidden = (error, context) => newError(StatusCodeHttp_1.StatusCodeHttp.FORBIDDEN, error, context);
exports.Forbidden = Forbidden;
const ServerError = (error, context) => newError(StatusCodeHttp_1.StatusCodeHttp.INTERNAL_SERVER_ERROR, error, context);
exports.ServerError = ServerError;
