// evio-library-connections/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
const shared_1 = require("./decorates/shared");
exports.default = {
    connect: shared_1.connect,
    close: shared_1.close,
    insertDocument: shared_1.insertDocument,
    findDocuments: shared_1.findDocuments,
    findOneDocument: shared_1.findOneDocument,
    findDocumentsByIdArray: shared_1.findDocumentsByIdArray,
    updateDocument: shared_1.updateDocument,
    deleteDocument: shared_1.deleteDocument,
    aggregateDocuments: shared_1.aggregateDocuments,
    upsertDocument: shared_1.upsertDocument,
    updateDocuments: shared_1.updateDocuments,
    findOneDocumentAndUpdate: shared_1.findOneDocumentAndUpdate,
    stringToObjectId: shared_1.stringToObjectId,
    bulkWriteDocuments: shared_1.bulkWriteDocuments,
    updateDocumentsWithOptions: shared_1.updateDocumentsWithOptions,
    findDocumentWithOptions: shared_1.findDocumentWithOptions,
    deleteDocuments: shared_1.deleteDocuments,
};
__exportStar(require("./classes/db.client"), exports);
