// evio-library-configs/dist/src/repository/appConfiguration.repository.js
"use strict";
var __classPrivateFieldGet = (this && this.__classPrivateFieldGet) || function (receiver, state, kind, f) {
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a getter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot read private member from an object whose class did not declare it");
    return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
};
var __classPrivateFieldSet = (this && this.__classPrivateFieldSet) || function (receiver, state, value, kind, f) {
    if (kind === "m") throw new TypeError("Private method is not writable");
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a setter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot write private member to an object whose class did not declare it");
    return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
};
var _a, _AppConfigurationRepository_instance, _AppConfigurationRepository_client, _AppConfigurationRepository_collectionName;
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppConfigurationReadRepository = void 0;
const configsDb_connection_1 = require("../common/configsDb.connection");
const appConfiguration_schema_1 = require("../schema/appConfiguration.schema");
class AppConfigurationRepository {
    constructor() {
        _AppConfigurationRepository_client.set(this, configsDb_connection_1.ConfigsDbConnection);
        _AppConfigurationRepository_collectionName.set(this, 'appconfigurations');
    }
    static get instance() {
        if (!__classPrivateFieldGet(_a, _a, "f", _AppConfigurationRepository_instance)) {
            __classPrivateFieldSet(_a, _a, new _a(), "f", _AppConfigurationRepository_instance);
        }
        return __classPrivateFieldGet(_a, _a, "f", _AppConfigurationRepository_instance);
    }
    async getAppConfigurationsByClient(clientName) {
        try {
            const document = await __classPrivateFieldGet(this, _AppConfigurationRepository_client, "f").findOne(__classPrivateFieldGet(this, _AppConfigurationRepository_collectionName, "f"), {
                clientName,
            });
            const wrapper = appConfiguration_schema_1.AppConfigurationSchema.safeParse(document);
            if (!wrapper.success) {
                console.log('not found or malformed app configuration', wrapper.error);
                return undefined;
            }
            return wrapper.data;
        }
        catch (error) {
            console.log('Error retrieving app configurations:', error);
            throw error;
        }
    }
    async getAppConfigurations() {
        try {
            const document = await __classPrivateFieldGet(this, _AppConfigurationRepository_client, "f").findMany(__classPrivateFieldGet(this, _AppConfigurationRepository_collectionName, "f"), {});
            console.log('document', document);
            return document;
        }
        catch (error) {
            console.log('Error retrieving app configurations:', error);
            throw error;
        }
    }
}
_a = AppConfigurationRepository, _AppConfigurationRepository_client = new WeakMap(), _AppConfigurationRepository_collectionName = new WeakMap();
_AppConfigurationRepository_instance = { value: void 0 };
exports.AppConfigurationReadRepository = AppConfigurationRepository.instance;
