// evio-library-configs/dist/src/repository/fee.repository.js
"use strict";
var __classPrivateFieldGet = (this && this.__classPrivateFieldGet) || function (receiver, state, kind, f) {
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a getter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot read private member from an object whose class did not declare it");
    return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
};
var __classPrivateFieldSet = (this && this.__classPrivateFieldSet) || function (receiver, state, value, kind, f) {
    if (kind === "m") throw new TypeError("Private method is not writable");
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a setter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot write private member to an object whose class did not declare it");
    return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
};
var _a, _FeeRepository_instance, _FeeRepository_client, _FeeRepository_collectionName;
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeeReadRepository = void 0;
const configsDb_connection_1 = require("../common/configsDb.connection");
const fee_schema_1 = require("../schema/fee.schema");
class FeeRepository {
    constructor() {
        _FeeRepository_client.set(this, configsDb_connection_1.ConfigsDbConnection);
        _FeeRepository_collectionName.set(this, 'fees');
    }
    static get instance() {
        if (!__classPrivateFieldGet(_a, _a, "f", _FeeRepository_instance)) {
            __classPrivateFieldSet(_a, _a, new _a(), "f", _FeeRepository_instance);
        }
        return __classPrivateFieldGet(_a, _a, "f", _FeeRepository_instance);
    }
    async findOneByCountryCode(countryCode) {
        const document = await __classPrivateFieldGet(this, _FeeRepository_client, "f").findOne(__classPrivateFieldGet(this, _FeeRepository_collectionName, "f"), {
            countryCode,
        });
        const wrapper = fee_schema_1.FeeSchema.safeParse(document);
        if (!wrapper.success) {
            console.log('not found or malformed fee', wrapper.error);
            return undefined;
        }
        return wrapper.data;
    }
    async findOneByCountryCodeAndZone(countryCode, zone) {
        const document = await __classPrivateFieldGet(this, _FeeRepository_client, "f").findOne(__classPrivateFieldGet(this, _FeeRepository_collectionName, "f"), {
            countryCode,
            zone,
        });
        const wrapper = fee_schema_1.FeeSchema.safeParse(document);
        if (!wrapper.success) {
            console.log('not found or malformed fee', wrapper.error);
            return undefined;
        }
        return wrapper.data;
    }
}
_a = FeeRepository, _FeeRepository_client = new WeakMap(), _FeeRepository_collectionName = new WeakMap();
_FeeRepository_instance = { value: void 0 };
exports.FeeReadRepository = FeeRepository.instance;
