// evio-library-configs/src/schema/appConfiguration.schema.ts
import { z } from 'zod';
import { MarketingAndPromotionNotifications } from '../enums';

// TODO: this schema should be used in the appConfiguration schema of evio-library-configs to avoid duplicate code
export const MapsConfigurationSchema = z.object({
  maxMap: z.number().min(0).default(4.5),  
  maxRankings: z.number().min(0).default(7),  
});

const MarketingAndPromotionNotificationsDefault = z.object({
  unsubscribedLink: z.string().default(MarketingAndPromotionNotifications.unsubscribedLink),
  licenseServiceEnabled: z.boolean().default(true),
  licenseProductEnabled: z.boolean().default(true),
  licenseMarketingEnabled: z.boolean().default(true),
}).default({});

export const LanguagesAllowedSchema = z.object({
  languages:  z.array(z.string()).optional(),
  default: z.string().optional(),
});

export const AppConfigurationSchema = z.object({
  clientName: z.string(),
  mapsConfiguration: MapsConfigurationSchema,
  marketingAndPromotionNotifications: MarketingAndPromotionNotificationsDefault,
  languagesAllowed: LanguagesAllowedSchema.optional(),
});



export type AppConfiguration = z.infer<typeof AppConfigurationSchema>;

export type LanguagesAllowed = z.infer<typeof LanguagesAllowedSchema>;
