// evio-library-configs/dist/src/schema/appConfiguration.schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppConfigurationSchema = exports.LanguagesAllowedSchema = exports.MapsConfigurationSchema = void 0;
const zod_1 = require("zod");
const enums_1 = require("../enums");
exports.MapsConfigurationSchema = zod_1.z.object({
    maxMap: zod_1.z.number().min(0).default(4.5),
    maxRankings: zod_1.z.number().min(0).default(7),
});
const MarketingAndPromotionNotificationsDefault = zod_1.z.object({
    unsubscribedLink: zod_1.z.string().default(enums_1.MarketingAndPromotionNotifications.unsubscribedLink),
    licenseServiceEnabled: zod_1.z.boolean().default(true),
    licenseProductEnabled: zod_1.z.boolean().default(true),
    licenseMarketingEnabled: zod_1.z.boolean().default(true),
}).default({});
exports.LanguagesAllowedSchema = zod_1.z.object({
    languages: zod_1.z.array(zod_1.z.string()).optional(),
    default: zod_1.z.string().optional(),
});
exports.AppConfigurationSchema = zod_1.z.object({
    clientName: zod_1.z.string(),
    mapsConfiguration: exports.MapsConfigurationSchema,
    marketingAndPromotionNotifications: MarketingAndPromotionNotificationsDefault,
    languagesAllowed: exports.LanguagesAllowedSchema.optional(),
});
