// evio-library-connections/dist/commons/manager.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.MongoDBConnectionManager = void 0;
const mongoose = __importStar(require("mongoose"));
const mongoose_1 = require("mongoose");
class MongoDBConnectionManager {
    constructor(uri, databaseName) {
        this._uri = uri;
        this._databaseName = databaseName;
        this.client = null;
        this.database = null;
    }
    async connect() {
        if (this.client && this.database && this.isConnectionValid()) {
            return this.database;
        }
        this.client = mongoose.createConnection(this._uri);
        this.database = this.client.useDb(this._databaseName);
        return this.database;
    }
    async close() {
        if (this.client) {
            await this.client.close();
            this.client = null;
            this.database = null;
        }
    }
    checkConnections() {
        if (this.client && !this.isConnectionValid()) {
            console.log('Connection state is different than connected.');
            this.close();
        }
    }
    isConnectionValid() {
        return this.client.readyState !== mongoose_1.ConnectionStates.connected;
    }
}
exports.MongoDBConnectionManager = MongoDBConnectionManager;
