// evio-library-commons/dist/src/enums/BillingEnums.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingDescription = exports.BillingStatus = void 0;
var BillingStatus;
(function (BillingStatus) {
    BillingStatus["UNPAID"] = "UNPAID";
    BillingStatus["PAID"] = "PAID";
    BillingStatus["INVOICED"] = "INVOICED";
    BillingStatus["INVOICED_PAID"] = "INVOICEDPAID";
    BillingStatus["UNKNOWN"] = "UNKNOWN";
    BillingStatus["FAILED_BILLING"] = "FAILEDBILLING";
    BillingStatus["MISSING_PAYMENT"] = "MISSINGPAYMENT";
    BillingStatus["CLOSED"] = "CLOSED";
    BillingStatus["ESTIMATION"] = "ESTIMATION";
    BillingStatus["NOT_PAID"] = "NOTPAID";
    BillingStatus["IN_BILLING"] = "INBILLING";
    BillingStatus["NOT_BILLABLE"] = "NOTBILLABLE";
})(BillingStatus || (exports.BillingStatus = BillingStatus = {}));
var BillingDescription;
(function (BillingDescription) {
    BillingDescription["UNKNOWN"] = "extraText_unknown";
    BillingDescription["ESTIMATION"] = "extraText_estimation";
    BillingDescription["NOT_PAID"] = "extraText_notPaid";
    BillingDescription["FAILED_BILLING"] = "extraText_failedBilling";
    BillingDescription["FAILED_BILLING_SUPPORT"] = "extraText_failedBillingSupport";
    BillingDescription["MISSING_PAYMENT"] = "extraText_missingPayment";
    BillingDescription["IN_BILLING"] = "extraText_inBilling";
    BillingDescription["IN_BILLING_PAID"] = "extraText_inBillingPaid";
    BillingDescription["NOT_BILLABLE"] = "extraText_notBillable";
})(BillingDescription || (exports.BillingDescription = BillingDescription = {}));
