// evio-library-commons/dist/src/enums/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StatusCodeHttp = exports.SalesTariffs = exports.OicpDeltaType = exports.PlugStatus = exports.AvailabilityType = exports.ParkingType = exports.PaginationRequestMode = exports.OicpParkingType = exports.OicpGeoCoordinatesType = exports.OicpEvseCapability = exports.OicpConnectorType = exports.OcpiEvseCapability = exports.ConnectorType = exports.ConnectorPowerType = exports.ConnectorFormat = exports.ChargerSubStatus = exports.ChargerStatus = exports.OcpiSessionSuspendedReason = exports.OcpiSessionStatus = exports.BillingDescriptions = exports.CardPhysicalState = exports.CardPhysicalPaymentState = exports.FilesUriType = exports.CardType = exports.AssetNetworkAction = exports.ContractStatus = exports.ContractTypes = exports.TokenStatus = exports.TokenTypes = exports.OcpiRoles = exports.OcpiModules = exports.PaymentsMethods = exports.PaymentStatus = exports.InvoiceStatus = exports.BillingDescription = exports.BillingStatus = exports.BillingInfo = exports.ThirdParties = exports.TariffTypes = exports.StationTypes = exports.DebtTransactionStatus = exports.BillingPeriods = exports.PaymentStatusSessions = exports.ChargerOperationStatuses = exports.PlatformCodes = exports.ChargerRankingFilters = exports.ChargerNetworks = exports.EVSEStatuses = exports.ClientNames = exports.ChargerTypes = void 0;
exports.CreateWaySessionsType = exports.DeviceTypes = exports.TinType = exports.PreAuthorizationStatusReason = exports.OcpiEnvironmentalImpact = exports.OcpiEnergySource = exports.PaymentStrategy = exports.OcpiFacilities = exports.OcpiParkingType = exports.OcpiConnectorType = exports.PaymentMethodStatus = exports.OperationalStatus = exports.SessionStatusesTextTypes = exports.SessionStatusesNumberTypes = exports.ChargerCommandResultType = exports.OcppWebSocketTypes = exports.SessionCommandResultType = exports.SessionCommandResponseType = exports.PaymentSubStatus = exports.PaymentType = exports.InvoiceLineCode = exports.TaxExemptionCode = exports.EmspTariffTypes = exports.SessionFlowLogsStatus = exports.SessionFlowLogsErrorTypes = exports.OicpStatusCodes = exports.OcppSessionStatuses = exports.OicpAuthorizationStatusType = exports.Networks = exports.DaysOfWeek = exports.OicpReferenceType = exports.OcpiTariffDimenstionType = exports.OcpiTariffType = exports.OicpReferenceUnit = exports.OicpPricingModel = exports.OicpEvseStatus = void 0;
const ChargerTypes_1 = require("./ChargerTypes");
Object.defineProperty(exports, "ChargerTypes", { enumerable: true, get: function () { return ChargerTypes_1.ChargerTypes; } });
const ClientNames_1 = require("./ClientNames");
Object.defineProperty(exports, "ClientNames", { enumerable: true, get: function () { return ClientNames_1.ClientNames; } });
const EVSEStatuses_1 = require("./EVSEStatuses");
Object.defineProperty(exports, "EVSEStatuses", { enumerable: true, get: function () { return EVSEStatuses_1.EVSEStatuses; } });
const ChargerNetworks_1 = require("./ChargerNetworks");
Object.defineProperty(exports, "ChargerNetworks", { enumerable: true, get: function () { return ChargerNetworks_1.ChargerNetworks; } });
const ChargerOperationStatuses_1 = require("./ChargerOperationStatuses");
Object.defineProperty(exports, "ChargerOperationStatuses", { enumerable: true, get: function () { return ChargerOperationStatuses_1.ChargerOperationStatuses; } });
const ChargerRankingFilters_1 = require("./ChargerRankingFilters");
Object.defineProperty(exports, "ChargerRankingFilters", { enumerable: true, get: function () { return ChargerRankingFilters_1.ChargerRankingFilters; } });
const PlatformCodes_1 = require("./PlatformCodes");
Object.defineProperty(exports, "PlatformCodes", { enumerable: true, get: function () { return PlatformCodes_1.PlatformCodes; } });
const PaymentStatusSessions_1 = require("./PaymentStatusSessions");
Object.defineProperty(exports, "PaymentStatusSessions", { enumerable: true, get: function () { return PaymentStatusSessions_1.PaymentStatusSessions; } });
const BillingPeriods_1 = require("./BillingPeriods");
Object.defineProperty(exports, "BillingPeriods", { enumerable: true, get: function () { return BillingPeriods_1.BillingPeriods; } });
const DebtTransactionStatus_1 = require("./DebtTransactionStatus");
Object.defineProperty(exports, "DebtTransactionStatus", { enumerable: true, get: function () { return DebtTransactionStatus_1.DebtTransactionStatus; } });
const StationTypes_1 = require("./StationTypes");
Object.defineProperty(exports, "StationTypes", { enumerable: true, get: function () { return StationTypes_1.StationTypes; } });
const TariffTypes_1 = require("./TariffTypes");
Object.defineProperty(exports, "TariffTypes", { enumerable: true, get: function () { return TariffTypes_1.TariffTypes; } });
const ThirdParties_1 = require("./ThirdParties");
Object.defineProperty(exports, "ThirdParties", { enumerable: true, get: function () { return ThirdParties_1.ThirdParties; } });
const PaymentStatus_1 = require("./PaymentStatus");
Object.defineProperty(exports, "PaymentStatus", { enumerable: true, get: function () { return PaymentStatus_1.PaymentStatus; } });
const BillingInfo_1 = require("./BillingInfo");
Object.defineProperty(exports, "BillingInfo", { enumerable: true, get: function () { return BillingInfo_1.BillingInfo; } });
const BillingEnums_1 = require("./BillingEnums");
Object.defineProperty(exports, "BillingStatus", { enumerable: true, get: function () { return BillingEnums_1.BillingStatus; } });
Object.defineProperty(exports, "BillingDescription", { enumerable: true, get: function () { return BillingEnums_1.BillingDescription; } });
const InvoiceStatus_1 = require("./InvoiceStatus");
Object.defineProperty(exports, "InvoiceStatus", { enumerable: true, get: function () { return InvoiceStatus_1.InvoiceStatus; } });
const PaymentsMethods_1 = require("./PaymentsMethods");
Object.defineProperty(exports, "PaymentsMethods", { enumerable: true, get: function () { return PaymentsMethods_1.PaymentsMethods; } });
const OcpiModules_1 = require("./OcpiModules");
Object.defineProperty(exports, "OcpiModules", { enumerable: true, get: function () { return OcpiModules_1.OcpiModules; } });
const OcpiRoles_1 = require("./OcpiRoles");
Object.defineProperty(exports, "OcpiRoles", { enumerable: true, get: function () { return OcpiRoles_1.OcpiRoles; } });
const TokenTypes_1 = require("./TokenTypes");
Object.defineProperty(exports, "TokenTypes", { enumerable: true, get: function () { return TokenTypes_1.TokenTypes; } });
const TokenStatus_1 = require("./TokenStatus");
Object.defineProperty(exports, "TokenStatus", { enumerable: true, get: function () { return TokenStatus_1.TokenStatus; } });
const ContractTypes_1 = require("./ContractTypes");
Object.defineProperty(exports, "ContractTypes", { enumerable: true, get: function () { return ContractTypes_1.ContractTypes; } });
const ContractStatus_1 = require("./ContractStatus");
Object.defineProperty(exports, "ContractStatus", { enumerable: true, get: function () { return ContractStatus_1.ContractStatus; } });
const AssetNetworkAction_1 = require("./AssetNetworkAction");
Object.defineProperty(exports, "AssetNetworkAction", { enumerable: true, get: function () { return AssetNetworkAction_1.AssetNetworkAction; } });
const CardTypes_1 = require("./CardTypes");
Object.defineProperty(exports, "CardType", { enumerable: true, get: function () { return CardTypes_1.CardType; } });
const FilesUriTypes_1 = require("./FilesUriTypes");
Object.defineProperty(exports, "FilesUriType", { enumerable: true, get: function () { return FilesUriTypes_1.FilesUriType; } });
const CardPhysicalPaymentState_1 = require("./CardPhysicalPaymentState");
Object.defineProperty(exports, "CardPhysicalPaymentState", { enumerable: true, get: function () { return CardPhysicalPaymentState_1.CardPhysicalPaymentState; } });
const CardPhysicalState_1 = require("./CardPhysicalState");
Object.defineProperty(exports, "CardPhysicalState", { enumerable: true, get: function () { return CardPhysicalState_1.CardPhysicalState; } });
const BillingDescriptions_1 = require("./BillingDescriptions");
Object.defineProperty(exports, "BillingDescriptions", { enumerable: true, get: function () { return BillingDescriptions_1.BillingDescriptions; } });
const OcpiSessionStatus_1 = require("./OcpiSessionStatus");
Object.defineProperty(exports, "OcpiSessionStatus", { enumerable: true, get: function () { return OcpiSessionStatus_1.OcpiSessionStatus; } });
const OcpiSessionSuspendedReason_1 = require("./OcpiSessionSuspendedReason");
Object.defineProperty(exports, "OcpiSessionSuspendedReason", { enumerable: true, get: function () { return OcpiSessionSuspendedReason_1.OcpiSessionSuspendedReason; } });
const ChargerStatus_1 = require("./ChargerStatus");
Object.defineProperty(exports, "ChargerStatus", { enumerable: true, get: function () { return ChargerStatus_1.ChargerStatus; } });
const ChargerSubStatus_1 = require("./ChargerSubStatus");
Object.defineProperty(exports, "ChargerSubStatus", { enumerable: true, get: function () { return ChargerSubStatus_1.ChargerSubStatus; } });
const ConnectorFormat_1 = require("./ConnectorFormat");
Object.defineProperty(exports, "ConnectorFormat", { enumerable: true, get: function () { return ConnectorFormat_1.ConnectorFormat; } });
const ConnectorPowerType_1 = require("./ConnectorPowerType");
Object.defineProperty(exports, "ConnectorPowerType", { enumerable: true, get: function () { return ConnectorPowerType_1.ConnectorPowerType; } });
const ConnectorType_1 = require("./ConnectorType");
Object.defineProperty(exports, "ConnectorType", { enumerable: true, get: function () { return ConnectorType_1.ConnectorType; } });
const OcpiEvseCapability_1 = require("./OcpiEvseCapability");
Object.defineProperty(exports, "OcpiEvseCapability", { enumerable: true, get: function () { return OcpiEvseCapability_1.OcpiEvseCapability; } });
const OicpConnectorType_1 = require("./OicpConnectorType");
Object.defineProperty(exports, "OicpConnectorType", { enumerable: true, get: function () { return OicpConnectorType_1.OicpConnectorType; } });
const OicpEvseCapability_1 = require("./OicpEvseCapability");
Object.defineProperty(exports, "OicpEvseCapability", { enumerable: true, get: function () { return OicpEvseCapability_1.OicpEvseCapability; } });
const OicpGeoCoordinatesType_1 = require("./OicpGeoCoordinatesType");
Object.defineProperty(exports, "OicpGeoCoordinatesType", { enumerable: true, get: function () { return OicpGeoCoordinatesType_1.OicpGeoCoordinatesType; } });
const OicpParkingType_1 = require("./OicpParkingType");
Object.defineProperty(exports, "OicpParkingType", { enumerable: true, get: function () { return OicpParkingType_1.OicpParkingType; } });
const PaginationRequestMode_1 = require("./PaginationRequestMode");
Object.defineProperty(exports, "PaginationRequestMode", { enumerable: true, get: function () { return PaginationRequestMode_1.PaginationRequestMode; } });
const ParkingType_1 = require("./ParkingType");
Object.defineProperty(exports, "ParkingType", { enumerable: true, get: function () { return ParkingType_1.ParkingType; } });
const AvailabilityType_1 = require("./AvailabilityType");
Object.defineProperty(exports, "AvailabilityType", { enumerable: true, get: function () { return AvailabilityType_1.AvailabilityType; } });
const PlugStatus_1 = require("./PlugStatus");
Object.defineProperty(exports, "PlugStatus", { enumerable: true, get: function () { return PlugStatus_1.PlugStatus; } });
const OicpDeltaType_1 = require("./OicpDeltaType");
Object.defineProperty(exports, "OicpDeltaType", { enumerable: true, get: function () { return OicpDeltaType_1.OicpDeltaType; } });
const StatusCodeHttp_1 = require("./StatusCodeHttp");
Object.defineProperty(exports, "StatusCodeHttp", { enumerable: true, get: function () { return StatusCodeHttp_1.StatusCodeHttp; } });
const SessionStatusesTypes_1 = require("./SessionStatusesTypes");
Object.defineProperty(exports, "SessionStatusesNumberTypes", { enumerable: true, get: function () { return SessionStatusesTypes_1.SessionStatusesNumberTypes; } });
Object.defineProperty(exports, "SessionStatusesTextTypes", { enumerable: true, get: function () { return SessionStatusesTypes_1.SessionStatusesTextTypes; } });
const SessionCommandResponseType_1 = require("./SessionCommandResponseType");
Object.defineProperty(exports, "SessionCommandResponseType", { enumerable: true, get: function () { return SessionCommandResponseType_1.SessionCommandResponseType; } });
const SessionCommandResultType_1 = require("./SessionCommandResultType");
Object.defineProperty(exports, "SessionCommandResultType", { enumerable: true, get: function () { return SessionCommandResultType_1.SessionCommandResultType; } });
const OicpEvseStatus_1 = require("./OicpEvseStatus");
Object.defineProperty(exports, "OicpEvseStatus", { enumerable: true, get: function () { return OicpEvseStatus_1.OicpEvseStatus; } });
const OicpPricingModel_1 = require("./OicpPricingModel");
Object.defineProperty(exports, "OicpPricingModel", { enumerable: true, get: function () { return OicpPricingModel_1.OicpPricingModel; } });
const OicpReferenceUnit_1 = require("./OicpReferenceUnit");
Object.defineProperty(exports, "OicpReferenceUnit", { enumerable: true, get: function () { return OicpReferenceUnit_1.OicpReferenceUnit; } });
const OcpiTariffType_1 = require("./OcpiTariffType");
Object.defineProperty(exports, "OcpiTariffType", { enumerable: true, get: function () { return OcpiTariffType_1.OcpiTariffType; } });
const OcpiTariffDimenstionType_1 = require("./OcpiTariffDimenstionType");
Object.defineProperty(exports, "OcpiTariffDimenstionType", { enumerable: true, get: function () { return OcpiTariffDimenstionType_1.OcpiTariffDimenstionType; } });
const OicpReferenceType_1 = require("./OicpReferenceType");
Object.defineProperty(exports, "OicpReferenceType", { enumerable: true, get: function () { return OicpReferenceType_1.OicpReferenceType; } });
const DaysOfWeek_1 = require("./DaysOfWeek");
Object.defineProperty(exports, "DaysOfWeek", { enumerable: true, get: function () { return DaysOfWeek_1.DaysOfWeek; } });
const Networks_1 = require("./Networks");
Object.defineProperty(exports, "Networks", { enumerable: true, get: function () { return Networks_1.Networks; } });
const SalesTariffs = __importStar(require("./SalesTariffs"));
exports.SalesTariffs = SalesTariffs;
const OicpAuthorizationStatusType_1 = require("./OicpAuthorizationStatusType");
Object.defineProperty(exports, "OicpAuthorizationStatusType", { enumerable: true, get: function () { return OicpAuthorizationStatusType_1.OicpAuthorizationStatusType; } });
const OcppSessionStatuses_1 = require("./OcppSessionStatuses");
Object.defineProperty(exports, "OcppSessionStatuses", { enumerable: true, get: function () { return OcppSessionStatuses_1.OcppSessionStatuses; } });
const OicpStatusCodes_1 = require("./OicpStatusCodes");
Object.defineProperty(exports, "OicpStatusCodes", { enumerable: true, get: function () { return OicpStatusCodes_1.OicpStatusCodes; } });
const SessionFlowLogsTypes_1 = require("./SessionFlowLogsTypes");
Object.defineProperty(exports, "SessionFlowLogsErrorTypes", { enumerable: true, get: function () { return SessionFlowLogsTypes_1.SessionFlowLogsErrorTypes; } });
Object.defineProperty(exports, "SessionFlowLogsStatus", { enumerable: true, get: function () { return SessionFlowLogsTypes_1.SessionFlowLogsStatus; } });
const EmspTariffTypes_1 = require("./EmspTariffTypes");
Object.defineProperty(exports, "EmspTariffTypes", { enumerable: true, get: function () { return EmspTariffTypes_1.EmspTariffTypes; } });
const TaxExemptionCode_1 = require("./TaxExemptionCode");
Object.defineProperty(exports, "TaxExemptionCode", { enumerable: true, get: function () { return TaxExemptionCode_1.TaxExemptionCode; } });
const InvoiceLineCode_1 = require("./InvoiceLineCode");
Object.defineProperty(exports, "InvoiceLineCode", { enumerable: true, get: function () { return InvoiceLineCode_1.InvoiceLineCode; } });
const PaymentType_1 = require("./PaymentType");
Object.defineProperty(exports, "PaymentType", { enumerable: true, get: function () { return PaymentType_1.PaymentType; } });
const PaymentSubStatus_1 = require("./PaymentSubStatus");
Object.defineProperty(exports, "PaymentSubStatus", { enumerable: true, get: function () { return PaymentSubStatus_1.PaymentSubStatus; } });
const OcppWebSocketTypes_1 = require("./OcppWebSocketTypes");
Object.defineProperty(exports, "OcppWebSocketTypes", { enumerable: true, get: function () { return OcppWebSocketTypes_1.OcppWebSocketTypes; } });
const ChargerCommandResultType_1 = require("./ChargerCommandResultType");
Object.defineProperty(exports, "ChargerCommandResultType", { enumerable: true, get: function () { return ChargerCommandResultType_1.ChargerCommandResultType; } });
const OperationalStatusEnum_1 = require("./OperationalStatusEnum");
Object.defineProperty(exports, "OperationalStatus", { enumerable: true, get: function () { return OperationalStatusEnum_1.OperationalStatus; } });
const PaymentMethodStatus_1 = require("./PaymentMethodStatus");
Object.defineProperty(exports, "PaymentMethodStatus", { enumerable: true, get: function () { return PaymentMethodStatus_1.PaymentMethodStatus; } });
const OcpiConnectorType_1 = require("./OcpiConnectorType");
Object.defineProperty(exports, "OcpiConnectorType", { enumerable: true, get: function () { return OcpiConnectorType_1.OcpiConnectorType; } });
const OcpiParkingType_1 = require("./OcpiParkingType");
Object.defineProperty(exports, "OcpiParkingType", { enumerable: true, get: function () { return OcpiParkingType_1.OcpiParkingType; } });
const OcpiFacilities_1 = require("./OcpiFacilities");
Object.defineProperty(exports, "OcpiFacilities", { enumerable: true, get: function () { return OcpiFacilities_1.OcpiFacilities; } });
const PaymentStrategyType_1 = require("./PaymentStrategyType");
Object.defineProperty(exports, "PaymentStrategy", { enumerable: true, get: function () { return PaymentStrategyType_1.PaymentStrategy; } });
const OcpiEnergySource_1 = require("./OcpiEnergySource");
Object.defineProperty(exports, "OcpiEnergySource", { enumerable: true, get: function () { return OcpiEnergySource_1.OcpiEnergySource; } });
const OcpiEnvironmentalImpact_1 = require("./OcpiEnvironmentalImpact");
Object.defineProperty(exports, "OcpiEnvironmentalImpact", { enumerable: true, get: function () { return OcpiEnvironmentalImpact_1.OcpiEnvironmentalImpact; } });
const PreAuthorizationStatusReason_1 = require("./PreAuthorizationStatusReason");
Object.defineProperty(exports, "PreAuthorizationStatusReason", { enumerable: true, get: function () { return PreAuthorizationStatusReason_1.PreAuthorizationStatusReason; } });
const TinType_1 = require("./TinType");
Object.defineProperty(exports, "TinType", { enumerable: true, get: function () { return TinType_1.TinType; } });
const DeviceTypes_1 = require("./DeviceTypes");
Object.defineProperty(exports, "DeviceTypes", { enumerable: true, get: function () { return DeviceTypes_1.DeviceTypes; } });
const CreateWaySessionsType_1 = require("./CreateWaySessionsType");
Object.defineProperty(exports, "CreateWaySessionsType", { enumerable: true, get: function () { return CreateWaySessionsType_1.CreateWaySessionsType; } });
__exportStar(require("./NetworkStatus"), exports);
__exportStar(require("./CardPhysicalStateInfo"), exports);
