// evio-library-commons/dist/src/enums/InvoiceLineCode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvoiceLineCode = void 0;
var InvoiceLineCode;
(function (InvoiceLineCode) {
    InvoiceLineCode["EnergyOutEmptyBt"] = "ISERV21001";
    InvoiceLineCode["EnergyEmptyBt"] = "ISERV21002";
    InvoiceLineCode["EnergyOutEmptyMt"] = "ISERV21003";
    InvoiceLineCode["EnergyEmptyMt"] = "ISERV21004";
    InvoiceLineCode["TarOutEmptyBt"] = "ISERV21005";
    InvoiceLineCode["TarEmptyBt"] = "ISERV21006";
    InvoiceLineCode["TarOutEmptyMt"] = "ISERV21007";
    InvoiceLineCode["TarEmptyMt"] = "ISERV21008";
    InvoiceLineCode["CpoFlat"] = "ISERV21009";
    InvoiceLineCode["CpoEnergy"] = "ISERV21010";
    InvoiceLineCode["CpoTime"] = "ISERV21011";
    InvoiceLineCode["Iec"] = "ISERV21012";
    InvoiceLineCode["Others"] = "ISERV21013";
    InvoiceLineCode["EvioMobilityServices"] = "ISERV21014";
    InvoiceLineCode["PublicGrant"] = "ISERV21019";
    InvoiceLineCode["MobieServices"] = "ISERV21023";
    InvoiceLineCode["OtherNetworks"] = "ISERV21024";
})(InvoiceLineCode || (exports.InvoiceLineCode = InvoiceLineCode = {}));
