// evio-library-commons/dist/src/enums/OcpiEvseCapability.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OcpiEvseCapability = void 0;
var OcpiEvseCapability;
(function (OcpiEvseCapability) {
    OcpiEvseCapability["ChargingProfileCapable"] = "CHARGING_PROFILE_CAPABLE";
    OcpiEvseCapability["ChargingPreferencesCapable"] = "CHARGING_PREFERENCES_CAPABLE";
    OcpiEvseCapability["ChipCardSupport"] = "CHIP_CARD_SUPPORT";
    OcpiEvseCapability["ContactlessCardSupport"] = "CONTACTLESS_CARD_SUPPORT";
    OcpiEvseCapability["CreditCardPayable"] = "CREDIT_CARD_PAYABLE";
    OcpiEvseCapability["DebitCardPayable"] = "DEBIT_CARD_PAYABLE";
    OcpiEvseCapability["PedTerminal"] = "PED_TERMINAL";
    OcpiEvseCapability["RemoteStartStopCapable"] = "REMOTE_START_STOP_CAPABLE";
    OcpiEvseCapability["Reservable"] = "RESERVABLE";
    OcpiEvseCapability["RfidReader"] = "RFID_READER";
    OcpiEvseCapability["TokenGroupCapable"] = "TOKEN_GROUP_CAPABLE";
    OcpiEvseCapability["UnlockCapable"] = "UNLOCK_CAPABLE";
})(OcpiEvseCapability || (exports.OcpiEvseCapability = OcpiEvseCapability = {}));
