// evio-library-commons/dist/src/enums/OcpiFacilities.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OcpiFacilities = void 0;
var OcpiFacilities;
(function (OcpiFacilities) {
    OcpiFacilities["HOTEL"] = "HOTEL";
    OcpiFacilities["RESTAURANT"] = "RESTAURANT";
    OcpiFacilities["CAFE"] = "CAFE";
    OcpiFacilities["MALL"] = "MALL";
    OcpiFacilities["SUPERMARKET"] = "SUPERMARKET";
    OcpiFacilities["SPORT"] = "SPORT";
    OcpiFacilities["RECREATION_AREA"] = "RECREATION_AREA";
    OcpiFacilities["NATURE"] = "NATURE";
    OcpiFacilities["MUSEUM"] = "MUSEUM";
    OcpiFacilities["BIKE_SHARING"] = "BIKE_SHARING";
    OcpiFacilities["BUS_STOP"] = "BUS_STOP";
    OcpiFacilities["TAXI_STAND"] = "TAXI_STAND";
    OcpiFacilities["TRAIN_STATION"] = "TRAIN_STATION";
    OcpiFacilities["AIRPORT"] = "AIRPORT";
    OcpiFacilities["PARKING_LOT"] = "PARKING_LOT";
    OcpiFacilities["CARPOOL_PARKING"] = "CARPOOL_PARKING";
    OcpiFacilities["FUEL_STATION"] = "FUEL_STATION";
    OcpiFacilities["WIFI"] = "WIFI";
    OcpiFacilities["TOILETS"] = "TOILETS";
    OcpiFacilities["SWIMMING_POOL"] = "SWIMMING_POOL";
    OcpiFacilities["HOSPITAL"] = "HOSPITAL";
    OcpiFacilities["POLICE"] = "POLICE";
    OcpiFacilities["HOTEL_RESTAURANT"] = "HOTEL_RESTAURANT";
})(OcpiFacilities || (exports.OcpiFacilities = OcpiFacilities = {}));
