// evio-library-commons/dist/src/enums/OicpConnectorType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OicpConnectorType = void 0;
var OicpConnectorType;
(function (OicpConnectorType) {
    OicpConnectorType["SmallPaddleInductive"] = "Small Paddle Inductive";
    OicpConnectorType["LargePaddleInductive"] = "Large Paddle Inductive";
    OicpConnectorType["AVCONConnector"] = "AVCON Connector";
    OicpConnectorType["TeslaConnector"] = "Tesla Connector";
    OicpConnectorType["NEMA520"] = "NEMA 5-20";
    OicpConnectorType["TypeEFrenchStandard"] = "Type E French Standard";
    OicpConnectorType["TypeFSchuko"] = "Type F Schuko";
    OicpConnectorType["TypeGBritishStandard"] = "Type G British Standard";
    OicpConnectorType["TypeJSwissStandard"] = "Type J Swiss Standard";
    OicpConnectorType["Type1ConnectorCableAttached"] = "Type 1 Connector (Cable Attached)";
    OicpConnectorType["Type2Outlet"] = "Type 2 Outlet";
    OicpConnectorType["Type2ConnectorCableAttached"] = "Type 2 Connector (Cable Attached)";
    OicpConnectorType["Type3Outlet"] = "Type 3 Outlet";
    OicpConnectorType["IEC60309SinglePhase"] = "IEC 60309 Single Phase";
    OicpConnectorType["IEC60309ThreePhase"] = "IEC 60309 Three Phase";
    OicpConnectorType["CCSCombo2PlugCableAttached"] = "CCS Combo 2 Plug (Cable Attached)";
    OicpConnectorType["CCSCombo1PlugCableAttached"] = "CCS Combo 1 Plug (Cable Attached)";
    OicpConnectorType["CHAdeMO"] = "CHAdeMO";
})(OicpConnectorType || (exports.OicpConnectorType = OicpConnectorType = {}));
