// evio-library-commons/dist/src/enums/OicpStatusCodes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OicpStatusCodes = void 0;
var OicpStatusCodes;
(function (OicpStatusCodes) {
    OicpStatusCodes["Success"] = "000";
    OicpStatusCodes["HubjectSystemError"] = "001";
    OicpStatusCodes["HubjectDatabaseError"] = "002";
    OicpStatusCodes["DataTransactionError"] = "009";
    OicpStatusCodes["UnauthorizedAccess"] = "017";
    OicpStatusCodes["InconsistentEvseId"] = "018";
    OicpStatusCodes["InconsistentEvcoId"] = "019";
    OicpStatusCodes["SystemError"] = "021";
    OicpStatusCodes["DataError"] = "022";
    OicpStatusCodes["QrCodeAuthFailedInvalidCredentials"] = "101";
    OicpStatusCodes["RfidAuthFailedInvalidUid"] = "102";
    OicpStatusCodes["RfidAuthFailedCardNotReadable"] = "103";
    OicpStatusCodes["PlcAuthFailedInvalidEvcoid"] = "105";
    OicpStatusCodes["NoPositiveAuthResponse"] = "106";
    OicpStatusCodes["QrCodeAppAuthFailedTimeout"] = "110";
    OicpStatusCodes["PlcAuthFailedInvalidUnderlyingEvcoid"] = "120";
    OicpStatusCodes["PlcAuthFailedInvalidCertificate"] = "121";
    OicpStatusCodes["PlcAuthFailedTimeout"] = "122";
    OicpStatusCodes["EvcoidLocked"] = "200";
    OicpStatusCodes["NoValidContract"] = "210";
    OicpStatusCodes["PartnerNotFound"] = "300";
    OicpStatusCodes["PartnerDidNotRespond"] = "310";
    OicpStatusCodes["ServiceNotAvailable"] = "320";
    OicpStatusCodes["SessionInvalid"] = "400";
    OicpStatusCodes["CommunicationToEvseFailed"] = "501";
    OicpStatusCodes["NoEvConnectedToEvse"] = "510";
    OicpStatusCodes["EvseAlreadyReserved"] = "601";
    OicpStatusCodes["EvseAlreadyInUseWrongToken"] = "602";
    OicpStatusCodes["UnknownEvseId"] = "603";
    OicpStatusCodes["EvseIdNotHubjectCompatible"] = "604";
    OicpStatusCodes["EvseOutOfService"] = "700";
})(OicpStatusCodes || (exports.OicpStatusCodes = OicpStatusCodes = {}));
