// evio-library-commons/dist/src/enums/SessionStatusesTypes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SessionStatusesNumberTypes = exports.SessionStatusesTextTypes = void 0;
var SessionStatusesTextTypes;
(function (SessionStatusesTextTypes) {
    SessionStatusesTextTypes["ACTIVE"] = "ACTIVE";
    SessionStatusesTextTypes["COMPLETED"] = "COMPLETED";
    SessionStatusesTextTypes["INVALID"] = "INVALID";
    SessionStatusesTextTypes["PENDING"] = "PENDING";
    SessionStatusesTextTypes["PENDING_STOP"] = "PENDING_STOP";
    SessionStatusesTextTypes["EXPIRED"] = "EXPIRED";
    SessionStatusesTextTypes["SUSPENDED"] = "SUSPENDED";
    SessionStatusesTextTypes["PENDING_DELAY"] = "PENDING_DELAY";
    SessionStatusesTextTypes["PENDING_START"] = "PENDING_START";
    SessionStatusesTextTypes["IN_PAUSE"] = "IN_PAUSE";
    SessionStatusesTextTypes["STOPPED_AND_EV_PARKED"] = "STOPPED_AND_EV_PARKED";
    SessionStatusesTextTypes["AVAILABLE_BUT_NOT_STOPPED"] = "AVAILABLE_BUT_NOT_STOPPED";
})(SessionStatusesTextTypes || (exports.SessionStatusesTextTypes = SessionStatusesTextTypes = {}));
;
var SessionStatusesNumberTypes;
(function (SessionStatusesNumberTypes) {
    SessionStatusesNumberTypes["ACTIVE"] = "20";
    SessionStatusesNumberTypes["COMPLETED"] = "40";
    SessionStatusesNumberTypes["INVALID"] = "60";
    SessionStatusesNumberTypes["PENDING"] = "10";
    SessionStatusesNumberTypes["PENDING_STOP"] = "30";
    SessionStatusesNumberTypes["EXPIRED"] = "60";
    SessionStatusesNumberTypes["SUSPENDED"] = "60";
    SessionStatusesNumberTypes["PENDING_DELAY"] = "90";
    SessionStatusesNumberTypes["PENDING_START"] = "100";
    SessionStatusesNumberTypes["IN_PAUSE"] = "50";
    SessionStatusesNumberTypes["STOPPED_AND_EV_PARKED"] = "70";
    SessionStatusesNumberTypes["AVAILABLE_BUT_NOT_STOPPED"] = "80";
})(SessionStatusesNumberTypes || (exports.SessionStatusesNumberTypes = SessionStatusesNumberTypes = {}));
;
