// evio-library-commons/dist/src/enums/SessionFlowLogsTypes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SessionFlowLogsErrorTypes = exports.SessionFlowLogsStatus = void 0;
var SessionFlowLogsStatus;
(function (SessionFlowLogsStatus) {
    SessionFlowLogsStatus["SUCCESS"] = "success";
    SessionFlowLogsStatus["ERROR"] = "error";
})(SessionFlowLogsStatus || (exports.SessionFlowLogsStatus = SessionFlowLogsStatus = {}));
var SessionFlowLogsErrorTypes;
(function (SessionFlowLogsErrorTypes) {
    SessionFlowLogsErrorTypes["VALIDATION_ERROR"] = "VALIDATION_ERROR";
    SessionFlowLogsErrorTypes["INTERNAL_ERROR"] = "INTERNAL_ERROR";
    SessionFlowLogsErrorTypes["EXTERNAL_SERVICE_ERROR"] = "EXTERNAL_SERVICE_ERROR";
    SessionFlowLogsErrorTypes["DEVICE_COMMUNICATION_ERROR"] = "DEVICE_COMMUNICATION_ERROR";
    SessionFlowLogsErrorTypes["TIMEOUT_ERROR"] = "TIMEOUT_ERROR";
    SessionFlowLogsErrorTypes["AUTHENTICATION_ERROR"] = "AUTHENTICATION_ERROR";
    SessionFlowLogsErrorTypes["UNKNOWN_ERROR"] = "UNKNOWN_ERROR";
})(SessionFlowLogsErrorTypes || (exports.SessionFlowLogsErrorTypes = SessionFlowLogsErrorTypes = {}));
