// evio-library-commons/dist/src/enums/TaxExemptionCode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxExemptionCode = void 0;
var TaxExemptionCode;
(function (TaxExemptionCode) {
    TaxExemptionCode["M01"] = "M01";
    TaxExemptionCode["M02"] = "M02";
    TaxExemptionCode["M03"] = "M03";
    TaxExemptionCode["M04"] = "M04";
    TaxExemptionCode["M05"] = "M05";
    TaxExemptionCode["M06"] = "M06";
    TaxExemptionCode["M07"] = "M07";
    TaxExemptionCode["M08"] = "M08";
    TaxExemptionCode["M09"] = "M09";
    TaxExemptionCode["M10"] = "M10";
    TaxExemptionCode["M11"] = "M11";
    TaxExemptionCode["M12"] = "M12";
    TaxExemptionCode["M13"] = "M13";
    TaxExemptionCode["M14"] = "M14";
    TaxExemptionCode["M15"] = "M15";
    TaxExemptionCode["M16"] = "M16";
    TaxExemptionCode["M19"] = "M19";
    TaxExemptionCode["M20"] = "M20";
    TaxExemptionCode["M21"] = "M21";
    TaxExemptionCode["M25"] = "M25";
    TaxExemptionCode["M26"] = "M26";
    TaxExemptionCode["M30"] = "M30";
    TaxExemptionCode["M31"] = "M31";
    TaxExemptionCode["M32"] = "M32";
    TaxExemptionCode["M33"] = "M33";
    TaxExemptionCode["M34"] = "M34";
    TaxExemptionCode["M40"] = "M40";
    TaxExemptionCode["M41"] = "M41";
    TaxExemptionCode["M42"] = "M42";
    TaxExemptionCode["M43"] = "M43";
    TaxExemptionCode["M99"] = "M99";
})(TaxExemptionCode || (exports.TaxExemptionCode = TaxExemptionCode = {}));
